python/fbprophet/diagnostics.py

```diff
@@ -82,9 +83,19 @@ def cross_validation(model, horizon, period=None, initial=None, multiprocess=Fal
         cross-validtation. If not provided works beginning from
         (end - horizon), works backwards making cutoffs with a spacing of period
         until initial is reached.
-    multiprocess: True, False, Optional (defaults to False). If `True`, use the
-        `multiprocessing` module to distribute each task to a different processor
-        core.
+    parallel : {None, 'processes', 'threads'}
+
+        How to parallelize the forecast computation. By default no parallelism
+        is used.
+
+        * None : No parallelism.
+        * 'processes' : Parallelize with concurrent.futures.ProcessPoolExectuor.
+        * 'threads' : Parallelize with concurrent.futures.ThreadPoolExecutor.
+            Note that some operations currently hold Python's Global Interpreter
```

Comment:
This may be worth looking into. In theory, thread-based parallelism can be faster since you don't need to serialize the (largeish) DataFrames between processes. But it seems like a decent amount of operations in `single_cutoff_forecast` and what calls seems to hold the GIL.

I'm not sure how feasible it is to improve this. Things like manipulating a DataFrame (slicing / adding columns, etc.) hold the GIL. Things like linear algebra don't.